### PR TITLE
Rework comments

### PR DESCRIPTION
**Pull request by Dionysusnu**

### CODE REVIEW

File: src/LuauAST/impl/create.ts [hunk 1]
```diff
@@ -90,7 +90,7 @@ export function id(name: string) {
  * @param text The text of the comment
  */
 export function comment(text: string) {
-	return luau.create(luau.SyntaxKind.Comment, { text: " " + text });
+	return luau.create(luau.SyntaxKind.Comment, { text });
```
Comment (osyrisrblx) [inline src/LuauAST/impl/create.ts:93]: This shorthand is meant for making simple single line comments (for things like macros, etc.). I don't think you should change this.

Instead, use `luau.create(luau.SyntaxKind.Comment, { text: "" })` where needed.

File: src/TSTransformer/util/getKindName.ts [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
-	if (result === "FirstStatement") return "VariableStatement";
-	if (result === "LastStatement") return "DebuggerStatement";
-	if (result === "FirstNode") return "QualifiedName";
-	if (result === "FirstJSDocNode") return "JSDocTypeExpression";
-	if (result === "FirstJSDocTagNode") return "JSDocTag";
-	if (result === "LastJSDocTagNode") return "JSDocPropertyTag";
-	if (result === "FirstContextualKeyword") return "AbstractKeyword";
-	if (result === "LastContextualKeyword") return "OfKeyword";
-	return result;
+export function getKindName(result: ts.SyntaxKind) {
```
Comment (osyrisrblx) [inline src/TSTransformer/util/getKindName.ts:3]: I don't really understand why these changes are necessary or even better than it was before?

Comment (osyrisrblx) [reply src/TSTransformer/util/getKindName.ts:3]: `result` no longer makes sense. If we were to go with these changes, we should rename it to `kind`.

File: src/TSTransformer/classes/TransformState.ts [hunk 3]
```diff
@@ -138,12 +138,17 @@ export class TransformState {
...
 		const commentRanges = ts.getLeadingCommentRanges(this.sourceFileText, node.pos) ?? [];
-		return (
-			commentRanges
-				// filter out non-`ts.SyntaxKind.SingleLineCommentTrivia`
-				.filter(commentRange => commentRange.kind === ts.SyntaxKind.SingleLineCommentTrivia)
-				// map each `ts.CommentRange` to its value without the beginning '//'
-				.map(commentRange => this.sourceFileText.substring(commentRange.pos + 2, commentRange.end))
+		return luau.list.make(
+			...commentRanges.map(commentRange =>
+				luau.comment(
```
Comment (osyrisrblx) [inline src/TSTransformer/classes/TransformState.ts:143]: Change this to luau.create call instead.

File: src/TSTransformer/util/getKindName.ts (hunk 2 above)
Comment (Dionysusnu) [reply src/TSTransformer/util/getKindName.ts:3]: Because ts.SyntaxKind is sometimes used without a `Node`.. experienced that while debugging this change. `CommentRange.kind`, for example

File: src/LuauAST/impl/create.ts (hunk 1 above)
Comment (Dionysusnu) [reply src/LuauAST/impl/create.ts:93]: It's taken care of in the comment renderer. It causes double indentation for block comments.

File: src/TSTransformer/classes/TransformState.ts (hunk 3 above)
Comment (Dionysusnu) [reply src/TSTransformer/classes/TransformState.ts:143]: What part of it?
